ProcessLib/ThermoHydroMechanics/CreateThermoHydroMechanicsProcess.cpp

```diff
@@ -0,0 +1,285 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateThermoHydroMechanicsProcess.h"
+
+#include <cassert>
+
+#include "MaterialLib/Fluid/FluidProperties/FluidProperties.h"
+#include "MaterialLib/Fluid/FluidProperties/CreateFluidProperties.h"
+#include "MaterialLib/SolidModels/CreateLinearElasticIsotropic.h"
+#include "MaterialLib/SolidModels/CreateLubby2.h"
+#include "MaterialLib/SolidModels/CreateEhlers.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+
+#include "ThermoHydroMechanicsProcess.h"
+#include "ThermoHydroMechanicsProcessData.h"
+
+namespace ProcessLib
+{
+namespace ThermoHydroMechanics
+{
+template <int DisplacementDim>
+class ThermoHydroMechanicsProcess;
+
+extern template class ThermoHydroMechanicsProcess<2>;
+extern template class ThermoHydroMechanicsProcess<3>;
+
+template <int DisplacementDim>
+std::unique_ptr<Process> createThermoHydroMechanicsProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "THERMO_HYDRO_MECHANICS");
+    DBUG("Create ThermoHydroMechanicsProcess.");
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    //! \ogs_file_param{prj__processes__process__THERMO_HYDRO_MECHANICS__material_property__fluid}
+    auto const& fluid_config = config.getConfigSubtree("fluid");
+    auto fluid_properties =
+        MaterialLib::Fluid::createFluidProperties(fluid_config);
```

Comment:
Move parsing of the fluid config after process variables.

Reply:
fixed